test/e2e/network/dns_common.go

```diff
@@ -474,7 +474,12 @@ func createProbeCommand(namesToResolve []string, hostEntries []string, ptrLookup
 
 	podARecByUDPFileName := fmt.Sprintf("%s_udp@PodARecord", fileNamePrefix)
 	podARecByTCPFileName := fmt.Sprintf("%s_tcp@PodARecord", fileNamePrefix)
-	probeCmd += fmt.Sprintf(`podARec=$$(getent hosts $$(hostname | awk '{print $1}') | tr ":." "-" | awk '{print $$1".%s.pod.%s"}');`, namespace, dnsDomain)
+
+	if framework.TestContext.IPFamily == "ipv4" {
+		probeCmd += fmt.Sprintf(`podARec=$$(hostname -i| awk -F. '{print $$1"-"$$2"-"$$3"-"$$4".%s.pod.%s"}');`, namespace, dnsDomain)
+	} else {
```

Comment:
kubernetes gives preference to IPv4, can we do the comparison the other way around? 
```
if framework.TestContext.IPFamily == "ipv6"` {
probeCmd ... ipv6
} else {
probeCmd ... ipv4
}

Reply:
Done